lib/dao.js

```diff
@@ -1628,18 +1636,28 @@ DataAccessObject.find = function find(query, options, cb) {
         } else {
           withNotify(item, next);
         }
-
+        
         function buildResult(data, callback) {
           const ctorOpts = {
             fields: query.fields,
             applySetters: false,
             persisted: true,
             // see https://github.com/strongloop/loopback-datasource-juggler/issues/1692
             applyDefaultValues: false,
+            // SPIKE: applyDefaultIdType to control reconversion to default type
           };
           let obj;
           try {
+            // SPIKE: This converts string-id back to ObjectID
             obj = new Model(data, ctorOpts);
+
+const propDefs = self.dataSource.definitions[self.modelName].properties;
+if (propDefs.id.type.name === 'ObjectID') {
+  // SPIKE: to change the constructor type
+  const strId = new String(obj.id.toString());
+  obj.id = strId;
+}
```

Comment:
Is it necessary to have an object-id specific coercion rule? Can we have a more generic rule that will coerce all non-string values to a string for properties defined with `type: string`? See https://github.com/strongloop/loopback-connector-mongodb/pull/591#discussion_r473908112

It's also important to coerce all properties, not just `id`:
1. The `id` property may not even exist, models can use a different name for the primary key.
2. We need to coerce foreign key values too (e.g. `Order.customerId`), otherwise query inclusion resolvers won't work.